Bugfix: use channel-defined middlewares (#131)
yiisoft/yii-queue: README.md

    or `MiddlewareConsumeInterface` depending on the place you use it.
- An array in the format of [yiisoft/definitions](https://github.com/yiisoft/definitions).
    **Only if you use yiisoft/definitions and yiisoft/di**.

> - A `callable`: `fn() => // do stuff`, `$object->foo(...)`, etc. It will be executed through the [yiisoft/injector](https://github.com/yiisoft/injector), so all the dependencies of your callable will be resolved

Reviewer: ```suggestion
- A `callable`: `fn() => // do stuff`, `$object->foo(...)`, etc. It will be executed through the
[yiisoft/injector](https://github.com/yiisoft/injector), so all the dependencies of your callable will be resolved.
```

Author: Why?

There is a restriction: by the time all middlewares are executed in the forward order, the adapter must be specified
in the `PushRequest` object. You will get a `AdapterNotConfiguredException`, if it isn't.

> You have three places to define push middlewares:

Reviewer: ```suggestion
You have three places to define push middlewares:

```

1. `PushMiddlewareDispatcher`. You can pass it either to the constructor, or to the `withMiddlewares()` method, which  

>     creates a completely new dispatcher object with only those middlewares, which are passed as arguments.

Reviewer: ```suggestion
creates a completely new dispatcher object with only those middlewares, which are passed as arguments.
```

> 2. Pass middlewares to the `Queue::withMiddlewares()` or `Queue::withMiddlewaresAdded()` method. The difference is that the first method will completely replace an existing middleware stack, while the second one will add passed middlewares to the end of the existing stack. These middlewares will be executed after the common ones, passed directly to the `PushMiddlewareDispatcher`. It's useful when defining a queue channel. Both methods return a new instance of the `Queue` class.
> 3. Put middlewares into the `Queue::push()` method like this: `$queue->push($message, ...$middlewares)`. These middlewares will be executed after those which are in the `PushMiddlewareDispatcher` and which are passed to the `Queue::withMiddlewares()` and `Queue::withMiddlewaresAdded()`, and only for the message with which they are passed.

Reviewer: ```suggestion
2. Pass middlewares to either `Queue::withMiddlewares()` or `Queue::withMiddlewaresAdded()` methods. The difference is 
that the former will completely replace an existing middleware stack, while the latter will add passed middlewares to 
the end of the existing stack. These middlewares will be executed after the common ones, passed directly to the 
`PushMiddlewareDispatcher`. It's useful when defining a queue channel. Both methods return a new instance of the `Queue` 
class.
3. Put middlewares into the `Queue::push()` method like this: `$queue->push($message, ...$middlewares)`. These
middlewares have the lowest priority and will be executed after those which are in the `PushMiddlewareDispatcher` and 
the ones passed to the `Queue::withMiddlewares()` and `Queue::withMiddlewaresAdded()` and only for the message passed 
along with them.
```
